src/azul/chalice.py

```diff
@@ -145,3 +161,160 @@ def _log_response(self, response):
     # Some type annotations to help with auto-complete
     lambda_context: LambdaContext
     current_request: Request
+
+
+class ValidatingAzulChaliceApp(AzulChaliceApp):
+
+    def __init__(self, app_name, unit_test=False, spec=None):
+        super().__init__(app_name=app_name, unit_test=unit_test, spec=spec)
+        self.route_specs: Mapping[str, RouteSpec] = {}
+
+    def route(self, *args, validate: bool = True, **kwargs):
+        """
+        Decorates a view handler function in a Chalice application. Refer to
+        docstring from super class for additional information.
+
+        :param validate: If False, do not validate requests submitted to the endpoint.
+        """
+
+        azul_chalice_decorator = super().route(*args, **kwargs)
+
+        def decorator(view_func):
+            @functools.wraps(view_func)
+            def f(*args, **kwargs):
+                if validate:
+                    self._validate()
+                return view_func(*args, **kwargs)
+
+            return azul_chalice_decorator(f)
+
+        return decorator
+
+    def _validate(self):
+        path = self.current_request.context['resourcePath']
+        self.route_specs[path].validate_params(self.current_request)
+
+    def _register_spec(self,
+                       path: str,
+                       path_spec: Optional[JSON],
+                       method_spec: Optional[JSON],
+                       methods: Iterable[str]):
+        """
+        Add a route's specifications to the specification object.
+        """
+        if path not in self.route_specs:
+            assert path not in self.specs['paths'], 'Only specify path_spec once per route path'
+            route_spec = RouteSpec(path, path_spec)
+            self.route_specs[path] = route_spec
```

Comment:
PyCharm shows a type warning here. Let me know if you need help resolving this. Make sure that there are no more type errors in this area after fixing this one.

Reply:
Changed self.route_specs to `MutableMapping` pycharm code inspection does not find any new warnings.